wallet/src/main/java/com/android/identity_credential/wallet/presentation/OnlinePresentmentActivity.kt

```diff
@@ -0,0 +1,820 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.lifecycleScope
+import com.android.identity.android.securearea.AndroidKeystoreKeyInfo
+import com.android.identity.android.securearea.AndroidKeystoreKeyUnlockData
+import com.android.identity.android.securearea.UserAuthenticationType
+import com.android.identity.cbor.Cbor
+import com.android.identity.cbor.CborArray
+import com.android.identity.cbor.Simple
+import com.android.identity.crypto.Algorithm
+import com.android.identity.document.Document
+import com.android.identity.document.DocumentRequest
+import com.android.identity.document.NameSpacedData
+import com.android.identity.issuance.CredentialFormat
+import com.android.identity.issuance.DocumentExtensions.documentConfiguration
+import com.android.identity.issuance.DocumentExtensions.issuingAuthorityIdentifier
+import com.android.identity.mdoc.credential.MdocCredential
+import com.android.identity.mdoc.mso.MobileSecurityObjectParser
+import com.android.identity.mdoc.mso.StaticAuthDataParser
+import com.android.identity.mdoc.response.DeviceResponseGenerator
+import com.android.identity.mdoc.response.DocumentGenerator
+import com.android.identity.mdoc.util.MdocUtil
+import com.android.identity.securearea.KeyLockedException
+import com.android.identity.securearea.KeyUnlockData
+import com.android.identity.trustmanagement.TrustPoint
+import com.android.identity.util.Constants
+import com.android.identity.util.Logger
+import com.android.identity_credential.wallet.R
+import com.android.identity_credential.wallet.WalletApplication
+import com.android.identity_credential.wallet.showBiometricPrompt
+import com.android.identity_credential.wallet.ui.ScreenWithAppBar
+import com.android.identity_credential.wallet.ui.destination.consentprompt.ConsentPrompt
+import com.android.identity_credential.wallet.ui.destination.consentprompt.ConsentPromptData
+import com.android.identity_credential.wallet.ui.theme.IdentityCredentialTheme
+import com.nimbusds.jose.EncryptionMethod
+import com.nimbusds.jose.JOSEObjectType
+import com.nimbusds.jose.JWEAlgorithm
+import com.nimbusds.jose.JWEHeader
+import com.nimbusds.jose.crypto.ECDHEncrypter
+import com.nimbusds.jose.jwk.ECKey
+import com.nimbusds.jose.jwk.JWKSet
+import com.nimbusds.jose.proc.DefaultJOSEObjectTypeVerifier
+import com.nimbusds.jose.proc.JWSKeySelector
+import com.nimbusds.jose.proc.SecurityContext
+import com.nimbusds.jose.shaded.gson.Gson
+import com.nimbusds.jose.util.Base64URL
+import com.nimbusds.jose.util.X509CertUtils
+import com.nimbusds.jwt.EncryptedJWT
+import com.nimbusds.jwt.JWTClaimsSet
+import com.nimbusds.jwt.SignedJWT
+import com.nimbusds.jwt.proc.BadJWTException
+import com.nimbusds.jwt.proc.DefaultJWTProcessor
+import com.nimbusds.jwt.proc.JWTClaimsSetVerifier
```

Comment:
Do we actually need to use this library? I mean, we already have some support for JOSE, see Algorithm.jwseAlgorithmIdentifier for example.... but maybe there's a lot of work we'd still have to reimplement?

Reply:
it's a bit intensive to replace, so it's better suited as a separate task - I added a TODO in the code for now

Reply:
OK, that's fine, thanks.